tests/Json/ValueSpec.php

```diff
@@ -712,26 +812,78 @@ public function it_should_fail_accessing_an_enum_if_the_enum_does_not_exist()
      * @test
      * @throws Exception\AssertionFailed
      */
-    public function it_should_provide_an_enum_if_accessing_a_non_null_optional_enum()
+    public function it_should_provide_a_vcn_enum_if_accessing_a_non_null_optional_vcn_enum(): void
     {
         $json = "A";
 
         $this->beConstructedWith($json, '$');
 
-        $this->¿enum(NonEmptyEnum::class)->shouldBe(NonEmptyEnum::A());
+        $this->¿enum(NonEmptyVcnEnum::class)->shouldBe(NonEmptyVcnEnum::A());
+    }
+
+    /**
+     * @test
+     * @throws Exception\AssertionFailed
+     */
+    public function it_should_provide_a_native_string_backed_enum_if_accessing_a_non_null_optional_native_string_backed_enum(): void
+    {
+        $json = "A";
+
+        $this->beConstructedWith($json, '$');
+
+        $this->¿enum(NonEmptyStringBackedEnum::class)->shouldBe(NonEmptyStringBackedEnum::A);
+    }
+
+    /**
+     * @test
+     * @throws Exception\AssertionFailed
+     */
+    public function it_should_provide_a_native_int_backed_enum_if_accessing_a_non_null_optional_native_int_backed_enum(): void
+    {
+        $json = "0";
+
+        $this->beConstructedWith($json, '$');
+
+        $this->¿enum(NonEmptyIntBackedEnum::class)->shouldBe(NonEmptyIntBackedEnum::ELT_ONE);
```

Comment:
This should fail? If an enum is backed by (serialized as) integers, why accept string input?

Reply:
fixed, but for least complex solution, it first considers if the value CAN be mapped and only then validates if the type is correct